http/src/main/java/io/crate/plugin/PipelineRegistry.java

```diff
@@ -88,22 +89,21 @@ public void addBefore(ChannelPipelineItem item) {
         }
     }
 
-    public void registerSslContext(SslContext sslContext) {
-        this.sslContext = sslContext;
-        if (sslContext != null) {
-            LOG.info("HTTP SSL support is enabled.");
-        } else {
-            LOG.info("HTTP SSL support is disabled.");
-        }
+    public void registerSslContext(Provider<SslContext> sslContextProvider) {
+        this.sslContextProvider = sslContextProvider;
     }
 
     public void registerItems(ChannelPipeline pipeline) {
         for (PipelineRegistry.ChannelPipelineItem item : addBeforeList) {
             pipeline.addBefore(item.base, item.name, item.handlerFactory.get());
         }
-        if (sslContext != null) {
-            SslHandler sslHandler = sslContext.newHandler(pipeline.channel().alloc());
-            pipeline.addFirst(sslHandler);
+
+        if (sslContextProvider != null) {
```

Comment:
The provider should never be null.

Reply:
I disagree as it's not a constructor argument. So I don't think being defensive hurts in this particular case.